forge-game/src/main/java/forge/game/trigger/TriggerPhaseOutAll.java

```diff
@@ -0,0 +1,44 @@
+package forge.game.trigger;
+
+import forge.game.ability.AbilityKey;
+import forge.game.card.Card;
+import forge.game.card.CardCollection;
+import forge.game.spellability.SpellAbility;
+import forge.util.Localizer;
+
+import java.util.Map;
+
+public class TriggerPhaseOutAll extends Trigger {
+
+    public TriggerPhaseOutAll(final Map<String, String> params, final Card host, final boolean intrinsic) {
+        super(params, host, intrinsic);
+    }
+
+    @Override
+    public boolean performTest(Map<AbilityKey, Object> runParams) {
+        final CardCollection cards = this.filterCards((CardCollection) runParams.get(AbilityKey.Cards));
+
+        return !cards.isEmpty();
+    }
+
+    @Override
+    public void setTriggeringObjects(SpellAbility sa, Map<AbilityKey, Object> runParams) {
+        final CardCollection filtered = this.filterCards((CardCollection) runParams.get(AbilityKey.Cards));
+
+        sa.setTriggeringObject(AbilityKey.Cards, filtered);
+    }
+
+    @Override
+    public String getImportantStackObjects(SpellAbility sa) {
+        StringBuilder sb = new StringBuilder();
+        sb.append(Localizer.getInstance().getMessage("lblPhasedOut")).append(": ");
+        sb.append(sa.getTriggeringObject(AbilityKey.Cards));
+        return sb.toString();
+    }
+
+    private CardCollection filterCards(CardCollection cards) {
+        cards.removeIf(card -> !matchesValidParam("ValidCards", card));
```

Comment:
matchesValidParam should already work fine with Iterable like CardCollection

Reply:
hmm, sorry I'm striking out on how to better set up this filter

Reply:
check `TriggerTokenCreatedOnce` for cleaner implementation

Reply:
😬 I see it yikes